app/controllers/characters_controller.rb

```diff
@@ -1,8 +1,7 @@
 # Character Controller
 class CharactersController < ApplicationController
-  before_action :authenticate_user!, except: [:show, :index]
```

Comment:
@zzokki81 this is more a question than anything else (I don't know the answer)... Will `load_and_authorize_resource` require user authentication too? I assume it does require it because you removed the `authenticate_user!` call.

Reply:
@zzokki81 you didn't respond to this question.

Reply:
Cancan is used for authorization  which is different from user authentication what devise gem does.
Authorization needs authentication..
So yeah, i think that load_and_authorize_resource require user authentication too..

Reply:
@zzokki81 in that case, did you remove it by mistake or...?

Reply:
Yes, my mistake :)